### PR TITLE
Tests around CSS specificity, and fix ordering in case of specificity clash

### PR DESCRIPTION
**Pull request by darrenburns**
When 2 rules have matching specificity, take the final one, not the first.

### CODE REVIEW

File: tests/css/test_stylesheet.py [hunk 1]
```diff
@@ -7,6 +7,98 @@
...
+
+def test_stylesheet_apply_highest_specificity_wins_multiple_classes():
+    """When we use two selectors containing only classes, then the selector
+    `.b.c` has greater specificity than the selector `.a`"""
+    css = ".b.c {background: blue;} .a {background: red; color: lime;}"
+    stylesheet = _make_stylesheet(css)
+    node = DOMNode(classes="a b c")
+    stylesheet.apply(node)
+
+    assert node.styles.background == Color(0, 0, 255)
```
Comment (DrBenton) [inline tests/css/test_stylesheet.py:49]: maybe using `COLOR_NAME_TO_RGB` would make these tests a bit easier to read? :slightly_smiling_face: 
```suggestion
    assert node.styles.background == COLOR_NAME_TO_RGB["blue"]
```

**Review by DrBenton — APPROVED** [review]
Nice to see these CSS selector specificity tests ^_^

File: tests/css/test_stylesheet.py (hunk 1 above)
Comment (darrenburns) [reply tests/css/test_stylesheet.py:49]: I did have `Color.parse("ansi_blue")` before but didn't think it added enough as the RGB tuple is pretty clear :D

File: src/textual/css/stylesheet.py [hunk 2]
```diff
@@ -297,13 +297,16 @@ def apply(self, node: DOMNode, animate: bool = False) -> None:
...
         # For each rule declared for this node, keep only the most specific one
         get_first_item = itemgetter(0)
-        node_rules: RulesMap = cast(
-            RulesMap,
-            {
-                name: max(specificity_rules, key=get_first_item)[1]
-                for name, specificity_rules in rule_attributes.items()
-            },
-        )
+        node_rules: RulesMap = cast(RulesMap, {})
```
Comment (willmcgugan) [inline src/textual/css/stylesheet.py:300]: Humor me. I suspect this is more work that required.

Can you restore this code, and change L291 to:

```python
    for rule in reversed(self.rules):
```

Comment (darrenburns) [reply src/textual/css/stylesheet.py]: That works, good spot!

**Review by willmcgugan — APPROVED** [review]

**Pull request merged**